### ISSUE TITLE
Using detectron2 to detect words in a dense document image

### ISSUE THREAD

**Issue by kawaii44** [facebookresearch/detectron2#779]
I'm trying to use detectron2/Mask-RCNN to find individual words in images of a historical UK Census.  After training a model for over a week, it does a decent job of masking out words. 
However, it still seems to be entirely missing most of the regions on the page.  An example:

![image](https://user-images.githubusercontent.com/60519095/73561386-22151e00-4416-11ea-93d0-f1f600fffa08.png)

This is what my config looks like:

```yaml
VERSION: 2
MODEL:
  META_ARCHITECTURE: "GeneralizedRCNN"
  BACKBONE:
    NAME: "build_resnet_fpn_backbone"
  RESNETS:
    OUT_FEATURES: ["res2", "res3", "res4", "res5"]
    STRIDE_IN_1X1: False  # this is a C2 model
    NUM_GROUPS: 32
    WIDTH_PER_GROUP: 8
    DEPTH: 101
  FPN:
    IN_FEATURES: ["res2", "res3", "res4", "res5"]
  ANCHOR_GENERATOR:
    SIZES: [[32], [64], [128], [256], [512]]
    ASPECT_RATIOS: [[0.5, 1.0, 2.0]]
  RPN:
    IN_FEATURES: ["p2", "p3", "p4", "p5", "p6"]
    PRE_NMS_TOPK_TRAIN: 2000
    PRE_NMS_TOPK_TEST: 10000
    POST_NMS_TOPK_TRAIN: 1000
    POST_NMS_TOPK_TEST: 5000
  ROI_HEADS:
    NAME: "StandardROIHeads"
    IN_FEATURES: ["p2", "p3", "p4", "p5"]
    SCORE_THRESH_TEST: 0.0
  ROI_BOX_HEAD:
    NAME: "FastRCNNConvFCHead"
    NUM_FC: 2
    POOLER_RESOLUTION: 7
  ROI_MASK_HEAD:
    NAME: "MaskRCNNConvUpsampleHead"
    NUM_CONV: 4
    POOLER_RESOLUTION: 14
  MASK_ON: True
  WEIGHTS: "detectron2://ImageNetPretrained/FAIR/X-101-32x8d.pkl"
  PIXEL_STD: [57.375, 57.120, 58.395]
DATASETS:
  TRAIN: ("word_finder_train",)
  TEST: ("word_finder_val",)
SOLVER:
  IMS_PER_BATCH: 2
  BASE_LR: 0.02
  STEPS: (210000, 250000)
  MAX_ITER: 270000
INPUT:
  MIN_SIZE_TRAIN: (640, 672, 704, 736, 768, 800)
```

So far, I've tried dialing down the minimum SCORE_THRESH_TEST to zero and bumping PRE_NMS_TOPK_TEST and POST_NMS_TOPK_TEST way up.  Still, it's missing most of the words on the page.   It's not even making a low confidence proposal for most of those words.

Simply put, my question is, what am I doing wrong here?  Any recommendations?

**Comment by ppwwyyxx** [comment]
My guess is that the number of outputs is limited by `TEST.DETECTIONS_PER_IMAGE`.

As the [issue template](https://github.com/facebookresearch/detectron2/blob/master/.github/ISSUE_TEMPLATE/unexpected-problems-bugs.md) mentions:

> If you expect the model to converge / work better, note that we do not give suggestions on how to train a new model. Only in one of the two conditions we will help with it: (1) You're unable to reproduce the results in detectron2 model zoo. (2) It indicates a detectron2 bug.

We provide configs & models with standard academic settings and expect users to have the knowledge to choose or design appropriate models & parameters for their own tasks.

**Comment by kawaii44** [comment]
Thanks for bending the rules for me ppwwyyxx.  You gave me exactly what I needed.  BTW: thanks for the great work you folks do on this platform!
